lib/matplotlib/artist.py

```diff
@@ -1161,12 +1135,53 @@ def properties(self):
         """Return a dictionary of all the properties of the artist."""
         return ArtistInspector(self).properties()
 
+    def _update_props(self, caller, props):
+        """Helper for `set` and `update`."""
+        errorfmt = {  # Often called & only internal, so use plain dict access.
+            "update": "{.__name__!r} object has no property {!r}",
+            "set": "{.__name__}.set() got an unexpected keyword argument {!r}",
+        }[caller]
+        ret = []
+        with cbook._setattr_cm(self, eventson=False):
+            for k, v in props.items():
+                # Allow attributes we want to be able to update through
+                # art.update, art.set, setp.
+                if k == "axes":
+                    ret.append(setattr(self, k, v))
+                else:
+                    func = getattr(self, f"set_{k}", None)
+                    if not callable(func):
+                        raise AttributeError(errorfmt.format(type(self), k))
+                    ret.append(func(v))
+        if ret:
+            self.pchanged()
+            self.stale = True
+        return ret
+
     def set(self, **kwargs):
         # docstring and signature are auto-generated via
         # Artist._update_set_signature_and_docstring() at the end of the
         # module.
-        kwargs = cbook.normalize_kwargs(kwargs, self)
-        return self.update(kwargs)
+        return self._update_props("set", cbook.normalize_kwargs(kwargs, self))
+
+    def _internal_update(self, **kwargs):
```

Comment:
I don't think there's any caller that passes individual args, so is there a need for the `**`?

Reply:
sure.